src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java

```diff
@@ -337,6 +358,37 @@ public int compare(FeedbackQuestionAttributes q1, FeedbackQuestionAttributes q2)
         return questions;
     }
     
+    /**
+     * Gets a {@code List} of all questions for the given session that
+     * the student can view/submit.
+     */
+    public List<FeedbackQuestionAttributes> getFeedbackQuestionsForStudent(
+            String feedbackSessionName, String courseId, StudentAttributes student) {
+
+        List<FeedbackQuestionAttributes> questions =
+                new ArrayList<FeedbackQuestionAttributes>();
+        
+        questions.addAll(
+                fqDb.getFeedbackQuestionsForGiverType(
+                        feedbackSessionName, courseId, FeedbackParticipantType.STUDENTS));
+        questions.addAll(
+                fqDb.getFeedbackQuestionsForGiverType(
+                        feedbackSessionName, courseId, FeedbackParticipantType.TEAMS));
+        
+        List<FeedbackQuestionAttributes> questionsWithCustomFeedbackPaths =
+                fqDb.getFeedbackQuestionsForGiverType(
+                        feedbackSessionName, courseId, FeedbackParticipantType.CUSTOM);
+        
+        for (FeedbackQuestionAttributes question : questionsWithCustomFeedbackPaths) {
+            if (question.hasStudentAsGiverInFeedbackPathAttributesList(student)) {
```

Comment:
While this should be fine too, you might consider using a bidirectional relationship between the path and question, or storing the question's Id/key in the feedback path, then the logic here maybe be clearer and possibly faster.